packages/machine-extractor/src/utils.ts

```diff
@@ -290,9 +308,12 @@ export const objectTypeWithKnownKeys = <
             let result: any | undefined;
 
             if (t.isObjectMethod(property.node)) {
-              result = parser.parse(property.node, context);
+              result = parser.parse(property.path, context);
             } else if (t.isObjectProperty(property.node)) {
-              result = parser.parse(property.node.value, context);
+              result = parser.parse(
+                (property.path as NodePath<t.ObjectProperty>).get("value"),
```

Comment:
That is a reason to prefer `property.isObjectProperty()`, based on that TS could refine `property` here and make this cast obsolete